jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -610,9 +619,20 @@ def export_extract(self, **kwargs):
             self.update_marks(step='ext')
 
         trace = self._get_ext_trace()
-        inp_sp2d = self._get_ext_input_spectrum().flux.value
+        inp_sp2d = self._get_ext_input_spectrum()
+
+        if self.ext_type_selected == 'Boxcar':
+            ext = extract.BoxcarExtract(inp_sp2d.data, trace, width=self.ext_width)
+        elif self.ext_type_selected == 'Horne':
+            uncert = inp_sp2d.uncertainty if inp_sp2d.uncertainty is not None else VarianceUncertainty(np.ones_like(inp_sp2d.data))  # noqa
+            if not hasattr(uncert, 'uncertainty_type'):
+                uncert = StdDevUncertainty(uncert)
+            image = NDData(inp_sp2d.data, uncertainty=uncert)
+            ext = extract.HorneExtract(image, trace)
```

Comment:
When are you thinking the `if not` clause will be triggered? I'm not aware of a case where the `uncertainty` attribute for either `CCDData` or `NDData` lacks an `uncertainty_type` attribute of its own.

Also, while it's allowed, specifying a `StdDevUncertainty`-type `uncertainty` currently triggers a warning from `HorneExtract` because the assumption is that people will want to pass variances instead.

Reply:
My question on `StdDevUncertainty` above still applies, but I think you may need to change the `if not` statement. For example, I get an `uncertainty_type` of `'unknown'` (not `None`) from the following:
```
NDData(np.arange(10), uncertainty=np.ones(10)).uncertainty.uncertainty_type
```

Reply:
`inp_sp2d` is a specutils Spectrum1D object, and I believe attaching an uncertainty is optional, and could be just an array (in which case `inp_sp2d.uncertainty` is an array which does not have the `uncertainty_type` attribute`).  So we need to decide fallbacks for at least two cases:
* when no uncertainty exists on the Spectrum1D object.  Specreduce would fallback on variances of all ones if not passing NDData, so I figured we should do the same here.
* when the uncertainty on the Spectrum1D object is an array (without a type).  This one is tricky.  You're right that specreduce will raise a warning if we assume `StdDevUncertainty`, but maybe a warning in this case is a good thing?  

And possibly your case, if that is possible to produce in a Spectrum1D object.  Hopefullly @rosteen can help give some insight from the specutils side of things.

Reply:
The uncertainty on a `Spectrum1D` is supposed to be a `NDUncertainty` object, not a bare array, but looking at the code I don't think there's actually a check to make sure the input uncertainty is an `NDUncertainty`. We should probably enforce that in `specutils`.

Reply:
I was debating whether it would be best to add ` or uncert.uncertainty_type == "unknown"` to the `hasattr` check, but honestly I think it's ok to have an error raised rather than making an assumption on the user's behalf.

Reply:
OK, I think the name/type combination for `inp_sp2d` confused me. I now understand it as a `Spectrum1D` object whose `flux` attribute is a 2D array of the image. Is that right?

Your second case is triggered by a scenario that isn't prohibited, but that we also don't want to encourage – using a bare array for a `Spectrum1D` object's `uncertainty` attribute. Are you saying that since `specutils` currently doesn't prohibit that scenario, you're using `StdDevUncertainty` on purpose to trigger the warning from `HorneExtract` as an indirect way to make the user think twice about providing that kind of input?

If so, I think it makes more sense to always assume `VarianceUncertainty` on the Jdaviz side for unclear uncertainties and to be more explicit about what we expect from the `Spectrum1D` object imported by the user. I guess this scenario becomes moot if @rosteen is able to enforce that all uncertainties must be in the `NDUncertainty` family, but there may be a reason why it's not that way.

Reply:
> I now understand it as a Spectrum1D object whose flux attribute is a 2D array of the image. Is that right?

Yes, that's right.

> Your second case is triggered by a scenario that isn't prohibited, but that we also don't want to encourage – using a bare array for a Spectrum1D object's uncertainty attribute. 

I agree - this isn't a use we should encourage, but since specutils _allows_ it, we have to handle it somehow (or catch it earlier in specutils or jdaviz and raise an error).

> Are you saying that since specutils currently doesn't prohibit that scenario, you're using StdDevUncertainty on purpose to trigger the warning from HorneExtract as an indirect way to make the user think twice about providing that kind of input?

A little bit of that but also somewhat assuming that people generally default to standard deviation (could be field-dependent though).  I would be a bit scared to default to something that doesn't raise a warning though (unless we raise our own warning).

> I guess this scenario becomes moot if @rosteen is able to enforce that all uncertainties must be in the NDUncertainty family, but there may be a reason why it's not that way.

That's true - @rosteen - if you think this would be a useful to have on specutils, should we create an issue/ticket there and then make a note to come back here and remove this if-statement once able?

Reply:
I think it should go into `specutils`, with the caveat that getting it in there might take anywhere between two weeks and three months 😬 

There's some old discussion of this [here](https://github.com/astropy/specutils/issues/529), I'd be happy to make this an upcoming priority in `specutils`. But that doesn't help us immediately for this PR.

Reply:
In that case, I think it would be good to issue a warning or notification about the standard deviation assumption when a `Spectrum1D` object with a numpy `array` uncertainty attribute is imported instead of during the export process. I'm also curious about whether standard deviation or variance is the best assumption to make in that case, but I don't have the context to know.

Reply:
I added an in-line plugin warning (below the dropdown where Horne is selected).  You can trigger it manually (for testing how it looks) with:

```
pext = specviz2d.app.get_tray_item_from_name('spectral-extraction')
pext.ext_uncert_warn = True
```

<img width="993" alt="image" src="https://user-images.githubusercontent.com/877591/188199951-5682d004-8a5e-41f7-b269-b9e68a40aef8.png">

Reply:
Could we include short instructions in the warning about how to remedy the issue if using standard deviations not OK? Since the `Spectrum1D` object they imported (`s1d`) should already have content in its `uncertainty` attribute, I believe they would just need to wrap `s1d.uncertainty.array` in a differently-typed `NDUncertainty` class, assign that as `s1d`'s uncertainty, and reimport `s1d` into Specviz2D.

Reply:
If the plugin gets much outside use, I don't think this scenario will be so rare. From past discussions with product owners about `HorneExtract`, it seems like many scientific users of objects like `CCDData`, `NDData`, `Spectrum1D`, etc. don't use the `NDUncertainty` family if they can get away with it because they're more used to lists and numpy arrays. Since @rosteen cautioned us against expecting the fix in `specutils` soon, I think it's good to have something to which we can point in the meantime.

I see what you're saying about the difficulty of knowing how the `Spectrum1D` object was loaded. Maybe this would provide more context: "Imported 2D image's uncertainties have unclear units; assuming standard deviation."

We could also add a short sentence in the part of `docs/specviz2d/import_data.rst` that discusses `Spectrum1D` saying that uncertainties are assumed to be standard deviations unless otherwise specified.

Reply:
I changed the warning text to: "WARNING: uncertainties on input 2D spectrum have unclear units; assuming standard deviation."

And added a note in the spectral extraction plugin docs (since for now that is the only place this assumption is made), saying: "Note: Horne extraction uses uncertainties on the input 2D spectrum.  If the spectrum uncertainties are not explicitly assigned a type, they are assumed to be standard deviation uncertainties."

Reply:
I think these are fair changes. I just suggested using the Sphinx's "note" directive for consistency with other pages, but the content looks good to me.